clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -7230,7 +7230,9 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   // -fuse-cxa-atexit is default.
   if (!Args.hasFlag(
           options::OPT_fuse_cxa_atexit, options::OPT_fno_use_cxa_atexit,
-          !RawTriple.isOSAIX() && !RawTriple.isOSWindows() &&
+          !RawTriple.isOSAIX() &&
+              (!RawTriple.isOSWindows() ||
+               RawTriple.isWindowsCygwinEnvironment()) &&
```

Comment:
mingw-w64 also has `__cxa_atexit` - should this be
```suggestion
               RawTriple.isOSCygMing()) &&
```
?  Or is there a good reason not to use `__cxa_atexit` on mingw-w64?

Reply:
As GCC doesn't use `__cxa_atexit` on mingw, we probably shouldn't change it unless we have good reason to do it. Best case it would be a simple uneventful change, worst case it would lead to a can of worms of breaking subtle things somewhere...